### DISCUSSION TITLE
Livewire bootstrap pagination view breaks non-livewire pagination usage

### DISCUSSION THREAD

**Discussion by rndb8418** [livewire/livewire#5130] [7. Bugs (archived)]
Hello,

The [bootstrap pagination view](https://github.com/livewire/livewire/blob/master/src/views/pagination/bootstrap.blade.php) breaks using pagination outside of Livewire in the project due to it's references of `$this` generating the following error:

```
Using $this when not in object context (View: vendor/livewire/livewire/src/views/pagination/bootstrap.blade.php) (View: 
vendor/livewire/livewire/src/views/pagination/bootstrap.blade.php) 
```

This page has no Livewire enabled components, but when using the standard eloquent pagination it stops working when trying to render the pagination via `$collection->links()` where `$collection` is the length aware paginated collection.

This seems inline with what the documentation says about Livewire pagination:

> Unfortunately, Livewire will overwrite a custom view you have defined inside a service provider using: Paginator::defaultView().

meaning any use of pagination in a project using Livewire will use the Livewire pagination view, and currently it appears to only work when used in the context of Livewire.

I'm able to resolve this by passing a view name to the `links()` call to use the Laravel provided views, however this is a rather tedious option as it requires remembering to add this in every pagination call that isn't Livewire.

Let me know if you need further details on the issue.

**Comment by joshhanley** [comment]
@rndb8418 yeah Livewire's pagination should only be used inside Livewire components. Outside you should still be using the standard Laravel pagination views.

But yeah as described, Livewire sets `Paginator::defaultView($this->paginationView());` to the Livewire pagination, that way you don't need to pass the view name into the links for all Livewire pagination.

This should only be impacted on a page that has a Livewire component that has the `WithPagination` trait included.

If a page doesn't have a Livewire component it shouldn't be impacted.

Hope this helps!

**Comment by rndb8418** [comment]
@joshhanley there's no reference/usage of livewire on this page that has the issue.

I've realised it's due to the system running on Laravel Octane, the issue only occurs if you visit a page that has Livewire features on prior to visiting the page that doesn't. If you go directly to the page without Livewire, the pagination works fine (until Livewire has been loaded).

It looks like Livewire's (or Laravels?) settings for the pagination change are being persisted between requests within Octane and we just need to add something to 'reset' those between requests. I can see there are [resolved issues relating to Octane & Livewire](https://github.com/livewire/livewire/pull/3987), perhaps pagination has been overlooked so far?

**Reply by joshhanley** [reply]
@rndb8418 ah yep Octane would do that! Hmm that's problematic.

Yeah I fixed a bunch previously.

I can fix this too, but we have an issue where we need to get the original pagination values so we can restore it.

I think we are going to need to cache your default somewhere first before Livewire changes it, so that way when Livewire cleans up after an Octane request, it can revert pagination back to the way it was before.

**Comment by joshhanley** [comment]
There is currently a work around for this here until we can get it fixed https://github.com/livewire/livewire/discussions/5351#discussioncomment-4091512

Hope this helps!
